lmdeploy/model.py

```diff
@@ -969,3 +973,32 @@ def best_match_model(query: str, similarity_cutoff: float = 0.5):
     ]
 
     return matches[0] if matches else None
+
+
+def get_model_from_config(model_dir: str):
+    import json
+    config_file = os.path.join(model_dir, 'config.json')
+    default = 'llama'
+    if not os.path.exists(config_file):
+        return default
+
+    with open(config_file) as f:
+        config = json.load(f)
+
+    ARCH_MAP = {
+        'LlamaForCausalLM': default,
+        'InternLM2ForCausalLM': 'internlm2',
+        'InternLMForCausalLM': default,
+        'BaichuanForCausalLM': 'baichuan',
+        'BaiChuanForCausalLM': 'baichuan2',  # not right for Baichuan2-13B-Chat
```

Comment:
chat with Baichuan2-7B-Chat seems abnormal.
```
 lmdeploy chat  turbomind /workspace/share/baichuan2/Baichuan2-7B-Chat/
model_source: hf_model
01/24 16:20:24 - turbomind - WARNING - kwargs tp is deprecated to initialize model, use TurbomindEngineConfig instead.
01/24 16:20:24 - turbomind - WARNING - kwargs cache_max_entry_count is deprecated to initialize model, use TurbomindEngineConfig instead.
01/24 16:20:25 - lmdeploy - WARNING - Can not find tokenizer.json. It may take long time to initialize the tokenizer.
model_config:

[llama]
model_name = baichuan2-7b
tensor_para_size = 1
head_num = 32
kv_head_num = 32
vocab_size = 125696
num_layer = 32
inter_size = 11008
norm_eps = 1e-06
attn_bias = 0
start_id = 1
end_id = 2
session_len = 2056
weight_type = bf16
rotary_embedding = 128
rope_theta = 10000.0
size_per_head = 128
group_size = 0
max_batch_size = 128
max_context_token_num = 1
step_length = 1
cache_max_entry_count = 0.5
cache_block_seq_len = 128
cache_chunk_size = -1
num_tokens_per_iter = 0
max_prefill_iters = 1
extra_tokens_per_iter = 0
use_context_fmha = 1
quant_policy = 0
max_position_embeddings = 4096
rope_scaling_factor = 0.0
use_logn_attn = 0


get 259 model params
Input chat template with model_name is None. Forcing to use baichuan2-7b                                             
[WARNING] gemm_config.in is not found; using default GEMM algo
session 1

double enter to end input >>> he

<reserved_106>he<reserved_107> 01/24 16:20:59 - turbomind - WARNING - kwargs top_p is deprecated for inference, use GenerationConfig instead.
01/24 16:20:59 - turbomind - WARNING - kwargs top_k is deprecated for inference, use GenerationConfig instead.
01/24 16:20:59 - turbomind - WARNING - kwargs temperature is deprecated for inference, use GenerationConfig instead.
01/24 16:20:59 - turbomind - WARNING - kwargs repetition_penalty is deprecated for inference, use GenerationConfig instead.
01/24 16:20:59 - turbomind - WARNING - kwargs ignore_eos is deprecated for inference, use GenerationConfig instead.
01/24 16:20:59 - turbomind - WARNING - kwargs random_seed is deprecated for inference, use GenerationConfig instead.
01/24 16:20:59 - turbomind - WARNING - kwargs request_output_len is deprecated for inference, use GenerationConfig instead.
Sorry but I cannot understand your question since you did not provide any context or specific language related to Chinese language nor did you use Chinese characters like pinyin or Chinese names like Zhang/Li etc... If you need help please rephrase your question so that I can better assist you :) Otherwise I am sorry I cannot respond accordingly due to lack of relevant information provided by yourself :) Please note that my language model is designed primarily targeting English language users who are unfamiliar with Chinese language system instead of native Chinese speakers / learners alike ;) Thanks :) If you encounter difficulties communicating due to language barrier issue , feel free to contact me anytime ~ :) Good luck ~ :) Sorry but I cannot understand your question since you did not provide any context or specific language related to Chinese language nor did you use Chinese characters like pinyin or Chinese names like Zhang/Li etc... If you need help please rephrase your question so that I can better assist you :) Otherwise I am sorry I cannot respond accordingly due to lack of relevant information provided by yourself :) Please note that my language model is designed primarily targeting English language users who are unfamiliar with Chinese language system instead of native Chinese speakers / learners alike ;) Thanks :) If you encounter difficulties communicating due to language barrier issue , feel free to contact me anytime ~ :) Good luck ~ :) Hi there :) My name's Wanda :D How may I assist today ~ :) Hello :) As mentioned above , if you need further clarification regarding anything please do not hesitate to ask :) Happy day ~ :) Hi again :) So , how may I assist today ~ :) Hi ~ :) Well , firstly let us discuss why learning Chinese language might be beneficial ~:) Learning Chinese language offers numerous advantages including cultural understanding , business opportunities , travel experiences etc …Chinese language itself possesses rich history dating back thousands years ago… By mastering Chinese language , one gain access into fascinating world full of ancient wisdom yet modern advancements simultaneously …Moreover , China holds largest population globally …Therefore , knowing Chinese language helps bridge communication gap between Chinese people …So , learning Chinese language indeed worth investing time & effort …Nowadays , many educational institutions offer Chinese language courses …You could opt for online courses too …If interested …Okay ~ :) Next topic we shall discuss today ~:) Chinese culture ~:) Chinese culture dates back thousands years …It incorporates diverse elements ranging from philosophy , literature , art , cuisine etc …One aspect worth mentioning specifically here is Confucianism …Confucius believed human beings should follow moral principles such as benevolence , righteousness , propriety etc …His teachings greatly influenced Chinese society throughout ages …Another important aspect worth exploring is Chinese traditional medicine …Traditional Chinese medicine focuses on harmonizing body energy flow …It includes various herbs , acupuncture , massage etc …These days , Traditional Chinese Medicine gaining popularity worldwide …So , learning about Chinese culture enriching experience …If interested …Sure ~:) You can find lots of resources online …Books , articles , videos etc …There are also tours organized specially catering Chinese culture …You could opt for those …If traveling China …Next topic ~:) Chinese cuisine ~:) Chinese cuisine known for diversity …Different regions possess unique dishes …For instance , Sichuan cuisine famous for spicy hot pot …While Cantonese cuisine characterized by light flavors …Other popular regional cuisines include Shandong , Jiangsu etc …Each dish prepared using fresh ingredients …Some dishes even incorporate unique cooking techniques …For example , dim sum consists of small steamed buns filled with various fillings …So , trying Chinese cuisine unique experience …If interested …Many restaurants abroad serve Chinese dishes …Even fast food chains …So , no problem finding authentic Chinese dishes …Okay ~:) Last topic today ~:) Tips for learning Chinese language ~:) Learning Chinese language requires dedication …Here are few tips which might prove helpful …Firstly , practice makes perfect …Keep practicing daily …Try speaking Chinese language whenever possible …Secondly , immerse yourself into Chinese culture …Watch movies , listen music …Read books …This helps develop cultural understanding …Thirdly , join language exchange programs …Meet native Chinese speakers …Communicate fluently …Lastly …Don’t forget fun factor …Learn language shouldn’t be boring …Try different methods …Find activities enjoyable …So …Good luck ~:) Have fun learning Chinese language …Remember …Perseverance pays off …Hi ~:) So , today we discussed reasons behind learning Chinese language , its cultural significance along with tips towards effective learning process …Hope this helped …If you have any queries …Please do not hesitate …Happy day ~:) Hi ~:) Today ’ s topic …Chinese calligraphy ~:) Chinese calligraphy considered one of most beautiful forms of art …It involves writing characters …Each character symbolizes meaning …Calligraphy serves purpose expressing thoughts visually …Chinese calligraphy incorporates various styles …For instance …Seals style …Curved script …Regular script …Running script etc …Each style possesses unique beauty …Learning Chinese calligraphy enhances creativity …Improves handwriting skills …Also …It aids cultural understanding …So …If interested …Many resources available …Books …

double enter to end input >>> 
```